stage1/init/common/pod.go

```diff
@@ -302,62 +302,13 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		return errwrap.Wrap(errors.New("unable to write environment file"), err)
 	}
 
-	var _uid, gid int
-	var err error
-
-	uidRange := uid.NewBlankUidRange()
-	if err := uidRange.Deserialize([]byte(privateUsers)); err != nil {
-		return errwrap.Wrap(errors.New("unable to deserialize uid range"), err)
-	}
-
-	if strings.HasPrefix(app.User, "/") {
-		var stat syscall.Stat_t
-		if err = syscall.Lstat(filepath.Join(common.AppRootfsPath(p.Root, appName),
-			app.User), &stat); err != nil {
-			return errwrap.Wrap(fmt.Errorf("unable to get uid from file %q",
-				app.User), err)
-		}
-		uidReal, _, err := uidRange.UnshiftRange(stat.Uid, 0)
-		if err != nil {
-			return errwrap.Wrap(errors.New("unable to determine real uid"), err)
-		}
-		_uid = int(uidReal)
-	} else {
-		_uid, err = strconv.Atoi(app.User)
-		if err != nil {
-			_uid, err = passwd.LookupUidFromFile(app.User,
-				filepath.Join(common.AppRootfsPath(p.Root, appName), "etc/passwd"))
-			if err != nil {
-				return errwrap.Wrap(fmt.Errorf("cannot lookup user %q", app.User), err)
-			}
-		}
-	}
-
-	if strings.HasPrefix(app.Group, "/") {
-		var stat syscall.Stat_t
-		if err = syscall.Lstat(filepath.Join(common.AppRootfsPath(p.Root, appName),
-			app.Group), &stat); err != nil {
-			return errwrap.Wrap(fmt.Errorf("unable to get gid from file %q",
-				app.Group), err)
-		}
-		_, gidReal, err := uidRange.UnshiftRange(0, stat.Gid)
-		if err != nil {
-			return errwrap.Wrap(errors.New("unable to determine real gid"), err)
-		}
-		gid = int(gidReal)
-	} else {
-		gid, err = strconv.Atoi(app.Group)
-		if err != nil {
-			gid, err = group.LookupGidFromFile(app.Group,
-				filepath.Join(common.AppRootfsPath(p.Root, appName), "etc/group"))
-			if err != nil {
-				return errwrap.Wrap(fmt.Errorf("cannot lookup group %q", app.Group), err)
-			}
-		}
+	Uid, Gid, err := parseUserGroup(p, ra, privateUsers)
```

Comment:
uid_, gid_ or u, g

On Tue, Apr 19, 2016 at 2:55 PM, Alban Crequy <notifications@github.com>
wrote:

> In stage1/init/common/pod.go
> <https://github.com/coreos/rkt/pull/2458#discussion_r60224605>:
>
> > -		}
> > -		_, gidReal, err := uidRange.UnshiftRange(0, stat.Gid)
> > -		if err != nil {
> > -			return errwrap.Wrap(errors.New("unable to determine real gid"), err)
> > -		}
> > -		gid = int(gidReal)
> > -	} else {
> > -		gid, err = strconv.Atoi(app.Group)
> > -		if err != nil {
> > -			gid, err = group.LookupGidFromFile(app.Group,
> > -				filepath.Join(common.AppRootfsPath(p.Root, appName), "etc/group"))
> > -			if err != nil {
> > -				return errwrap.Wrap(fmt.Errorf("cannot lookup group %q", app.Group), err)
> > -			}
> > -		}
> > +	Uid, Gid, err := parseUserGroup(p, ra, privateUsers)
>
> just curious, does Golang advise an upper-case first letter to avoid
> conflicts with package name, or did you just decide it?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rkt/pull/2458/files/b47f720f4d7e217ea085bc14065130b40564392f#r60224605>
>